Clarify command_line runtime environment

LarsMichelsen opened the pull request:
A user should know in which environment his commands are executed and how the results of the command are processed.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

When using the integration for the first time I was a little lost because I did not know in which kind of environment my command is executed and how the results are processed. After trying out a bit and reading the code, it was clearer to me. I hope I got that right.

Now I think this would be useful to others starting with this integration.

To be honest, I wasn't quite sure where the text was properly located.

1. It is not clear to me whether you can simply open any chapters in the articles.
2. The details actually apply to all integrations of the component. I have now described it for sensor and binary_sensor. Should it be described for the others (cover, switch?) as well?
3. The copy & paste of the information is actually not good. Is there a central place where this can or should be documented?

- Link to parent pull request in the codebase: -
- Link to parent pull request in the Brands repository: -
- This PR fixes or closes issue: -

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

LarsMichelsen committed:
Clarify command_line runtime environment

A user should know in which environment his commands are executed
and how the results of the command are processed.

homeassistant commented:
Hi @LarsMichelsen, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#19150).

Once you do that we will be able to review and accept this pull request.

Thanks!

LarsMichelsen committed:
Clarify shell_command runtime environment

A user should know in which environment his commands are executed
and how the results of the command are processed.

frenck commented on source/_integrations/command_line.markdown:
```diff
@@ -74,6 +74,16 @@ command_timeout:
+The `command` is executed within the [configuration directory](/docs/configuration/).
+
+<div class='note'>
+In a dockerized environment, the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
```
I think we should avoid "dockerized" environment for several reasons:
- One doesn't have to be using Docker
- Unclear for people not aware of containerization (e.g., when using the Home Assistant operating system). These people won't be aware this comment applies to them.

LarsMichelsen replied:
Thanks for your review.

I can clearly understand this point of view. However, when first working with the integration I was lost and had to debug myself through the logic to understand it better. For me this worked, but I guess it will not work for everyone.

This is why I proposed to add these information to the documentation. I intentionally put the docker information into a "note" block to mark it as additional information which may be useful in some specific situation.

Using "Home Assistant operating system" is fine and nice. I use it myself. This abstraction is cool and works very well for basic tasks and so on. But if you want to execute commands, you are imho leaving these basic tasks and will have to know which environment is used to understand things.

If it's only about the wording I will happily change it to something like: "If you are using 'Home Assistant operating system' or Home assistant in a docker container, the commands are executed in a container which is running the `homeassistant` image". Or something like this. I am open to suggestions.

I wanted to bring this up again to improve the text. If this should not convince you, then I take the block also gladly out. It is then still more helpful than it was before.

Let me know what you think.

frenck replied:
I was not declining the suggestion, I was saying the current suggestion is confusing and entirely correct. Could you adjust the text in a way that it resolves my review comments? Thanks 👍

LarsMichelsen replied:
Okay, no problem. I just added an update.

frenck commented on source/_integrations/command_line.markdown:
```diff
@@ -74,6 +74,16 @@ command_timeout:
+
+<div class='note'>
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
+</div>
```
```suggestion
<div class='note'>

If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.

</div>
```

frenck committed:
Tweak

frenck commented on source/_integrations/sensor.command_line.markdown:
```diff
@@ -56,6 +56,16 @@ json_attributes:
+
+<div class='note'>
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
+</div>
```
```suggestion
<div class='note'>

If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.

</div>
```

frenck committed:
Tweak

frenck commented on source/_integrations/sensor.command_line.markdown:
```diff
@@ -56,6 +56,16 @@ json_attributes:
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
+</div>
+
+With a `0` exit code, the output (stdout) of the command is used as `value`. In case a command results in a non `0` exit code or is terminated by the `command_timeout`, the result is only logged to Home Assistant log and the sensors value is not updated.
```
```suggestion
With a `0` exit code, the output (stdout) of the command is used as `value`. In case a command results in a non `0` exit code or is terminated by the `command_timeout`, the result is only logged to Home Assistant log and the value of the sensor is not updated.
```

frenck committed:
Tweak

frenck commented on source/_integrations/shell_command.markdown:
```diff
@@ -39,6 +39,16 @@ Any service data passed into the service call to activate the shell command will
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
+</div>
+
+With a `0` exit code, the output (stdout) of the command is used as `value`. In case a command results in a non `0` exit code or is terminated by the `command_timeout`, the result is only logged to Home Assistant log and the sensors value is not updated.
```
```suggestion
With a `0` exit code, the output (stdout) of the command is used as `value`. In case a command results in a non `0` exit code or is terminated by the `command_timeout`, the result is only logged to Home Assistant log and the value of the sensor is not updated.
```

frenck committed:
Tweak

frenck commented on source/_integrations/command_line.markdown:
```diff
@@ -74,6 +74,18 @@ command_timeout:
+
+<div class='note'>
+
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
```
```suggestion
If you are using [Home Assistant Operating System](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
```

frenck committed:
Tweak

frenck commented on source/_integrations/sensor.command_line.markdown:
```diff
@@ -56,6 +56,18 @@ json_attributes:
+
+<div class='note'>
+
+If you are using [Home Assistant operating system](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
```
```suggestion
If you are using [Home Assistant Operating System](https://github.com/home-assistant/operating-system), the commands are executed in the `homeassistant` container context. So if you test or debug your script, it might make sense to do this in the context of this container to get the same runtime environment.
```

frenck committed:
Tweak
